pdnshttp/client_test.go

```diff
@@ -17,7 +17,7 @@ func TestGetExecutedCorrectly(t *testing.T) {
 		JSON(map[string]string{"foo": "bar"})
 
 	hc := &http.Client{Transport: gock.DefaultTransport}
-	c := NewClient("http://test.example", hc, &APIKeyAuthenticator{APIKey: "secret"}, ioutil.Discard)
+	c := NewClient("http://test.example", hc, &APIKeyAuthenticator{APIKey: "secret"}, io.Discard)
```

Comment:
Maybe we could also add a second test case (next to `TestGetExecutedCorrectly`) that tests the new baseURL behaviour? So, basically just by coping the `TestGetExecutedCorrectly` with different base URLs?

Reply:
I've added a test case. There is something weird though the way that gock works, since it doesn't seem to be very strict with the check. Seems to be some kind of a contains check. I was also able to break it if I changed the v1 and v2 values, but ideally the assertion would be a bit stricter imo.

Reply:
That's weird; 🤔  I'm not overly familiar with gock but I always assumed that it was fairly strict when doing assertions. In any case, the test cases look fine to me as they are. 👍